superset/assets/src/query/buildDatasource.ts

```diff
@@ -0,0 +1,19 @@
+import { FormData } from './formData';
+
+enum DatasourceType {
+  Table = 'table',
+  Druid = 'druid',
+}
+
+export interface Datasource {
+  id: number;
+  type: DatasourceType;
+}
+
+export default function buildDatasource(formData: FormData): Datasource {
+  const [id, type] = formData.datasource.split('__');
+  return {
+    id: parseInt(id, 10),
+    type: type === 'table' ? DatasourceType.Table : DatasourceType.Druid,
```

Comment:
Is this line useful? Also if someone in the wild has a custom connector it would break here.